test/e2e/node_exporter_test.go

```diff
@@ -254,3 +254,60 @@ nodeExporter:
 	}
 
 }
+
+func TestNodeExporterNetworkDevicesExclusion(t *testing.T) {
+	t.Cleanup(func() {
+		f.MustDeleteConfigMap(t, f.BuildCMOConfigMap(t, ""))
+	})
+
+	tests := []struct {
+		name      string
+		config    string
+		filter    string
+		predicate func(float64) bool
+	}{
+		{
+			name:      "default devices generate metrics",
+			predicate: func(f float64) bool { return f > 0 },
+		},
+		{
+			name:      "default devices include 'lo'",
+			filter:    `device="lo"`,
+			predicate: func(f float64) bool { return f > 0 },
+		},
+		{
+			name: "excluding 'lo'",
+			config: `
+nodeExporter:
+  ignoredNetworkDevices:
+  - lo`,
+			filter:    `device="lo"`,
+			predicate: func(f float64) bool { return f == 0 },
```

Comment:
when excluding devices, the query needs to use absent() otherwise it could see metrics from the past.

Reply:
yep I figured that out when I was finally able to test locally :-)